app/src/org/commcare/dalvik/activities/SingleAppManagerActivity.java

```diff
@@ -0,0 +1,312 @@
+package org.commcare.dalvik.activities;
+
+import org.commcare.android.database.global.models.ApplicationRecord;
+import org.commcare.android.util.SessionUnavailableException;
+import org.commcare.dalvik.R;
+import org.commcare.dalvik.application.CommCareApp;
+import org.commcare.dalvik.application.CommCareApplication;
+import org.commcare.dalvik.services.CommCareSessionService;
+import org.javarosa.core.services.locale.Localization;
+
+import android.app.Activity;
+import android.app.AlertDialog;
+import android.content.DialogInterface;
+import android.content.Intent;
+import android.content.SharedPreferences;
+import android.os.Bundle;
+import android.view.View;
+import android.widget.Button;
+import android.widget.TextView;
+import android.widget.Toast;
+
+/**
+ * The activity launched when the user clicks on a specific app within the app manager. From
+ * this screen, the selected app can be archived/unarchived, uninstalled, updated, or have its
+ * multimedia verified if needed.
+ *
+ * @author amstone
+ */
+
+public class SingleAppManagerActivity extends Activity {
+
+    private ApplicationRecord appRecord;
+    private AlertDialog dialog;
+    private static final int LOGOUT_FOR_UPDATE = 0;
+    private static final int LOGOUT_FOR_VERIFY_MM = 1;
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.single_app_view);
+        // Try to retrieve the app record at the indicated position
+        int position = getIntent().getIntExtra("position", -1);
+        appRecord = CommCareApplication._().getAppAtIndex(position);
+        // Implies that this appRecord has been uninstalled since last we launched
+        // SingleAppManagerActivity, so redirect to AppManagerActivity
+        if (appRecord == null) {
+            Intent i = new Intent(getApplicationContext(), AppManagerActivity.class);
+            startActivity(i);
+            finish();
+        }
+        //Set app name
+        String appName = appRecord.getDisplayName();
+        TextView tv = (TextView)findViewById(R.id.app_name);
```

Comment:
I think it would also be helpful to display the App Version somewhere in this metadata.